docs/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,192 @@
+
+   For production use, we recommend that your MongoDB deployment use valid
+   certificates generated and signed by the same certificate authority.
+   For testing, your deployment can use self-signed certificates.
+
+The following list describes the components that your client must
+present to establish a TLS-enabled connection:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - TLS Component
+     - Description
+
+   * - Certificate Authority (CA)
+     - One or more certificate authorities to
+       trust when making a TLS connection.
+
+   * - Client Certificate
+     - A digital certificate that allows the server to verify the identity
+       of your application to establish an encrypted network connection.
+
+   * - Certificate Key
+     - The client certificate private key file. This key is often
+       included within the certificate file itself.
+
+   * - Passphrase
+     - The password to decrypt the private client key if it is encrypted.
+
+Reference Certificates
+----------------------
+
+You must reference your certificates when configuring your ``mongodb``
+connection so that the server can validate them before the client connects.
+
+We recommend that you reference your certificates and set other TLS
+options in the ``options`` property of your connection configuration
+instead of in the connection string. This improves readability and
+flexibility in your application.
+
+Set the following options in the ``options`` property to reference your
+certificates:
+
+- ``tlsCAFile``
+- ``tlsCertificateKeyFile``
+- ``tlsCertificateKeyFilePassword``
+
+You can specify additional options to configure TLS on your connection.
+For **testing purposes**, you can set the
+``tlsAllowInvalidCertificates``, ``tlsAllowInvalidHostnames``, or
+``tlsInsecure`` fields to ``true``.
+   
+.. warning::
+
+   Setting these options to ``true`` disables
+   certificate and hostname validation.
+   
+   Specifying these options in a production environment makes
+   your application insecure and potentially
+   vulnerable to expired certificates and foreign processes posing
+   as valid client instances.
+
+The following example configures a connection with TLS enabled:
+
+.. code-block:: php
+
+   'connections' => [
+   
+       'mongodb' => [
+           'driver' => 'mongodb',
+           'dsn' => '<connection string>',
+           'database' => 'myDB',
+           'options'  => [
+               'tls' => true,
+               'tlsCAFile' => '<path to CA certificate>',
+               'tlsCertificateKeyFile' => '<path to public client certificate>',
+           ],
+       ], ...
+   ]
+
+.. note::
+
+   You can alternatively reference your certificates by setting options
+   in the ``driverOptions`` property in your connection configuration.
+   To learn more about these options, see the
+   `MongoDB\Driver\Manager::__construct()
+   <https://www.php.net/manual/en/mongodb-driver-manager.construct.php>`__
+   API documentation.
```

Comment:
Suggestion:

In the referenced link, it's not clear to me which settings are safe to use. The driverOptions settings, such as such as ca_file, pem_file, and pem_pwd, on the page are described as "deprecated alias for the [option name] URI option" . 

I think it could be helpful to mention which settings to recommend since it's not clear on the linked page. Maybe @GromNaN can provide more guidance here.

Reply:
I refer you to @jmikola. I don't know this option. I learned about this feature by reading your doc.

Reply:
The URI options prefixed with `tls` supersede the `driverOptions`. There are very few options that are _only_ supported through `driverOptions` (e.g. `crl_file`), and I don't think any of those are relevant to these docs.

I think the right call is to remove any discussion of `driverOptions`. You can still link to the [Manager constructor](https://www.php.net/manual/en/mongodb-driver-manager.construct.php) as canonical documentation for URI options, though. In that case, you can drop a link to it in the "Additional Information" section below (probably _above_ the server docs, since it will be more relevant to PHP).

----

> ``driverOptions`` property

Is this actually a property in the Laravel configuration? If you're referring to the Manager constructor, then the proper terminology would be "`driverOptions` _parameter_" (not "property").

But this may be a moot point since the entire note can be removed.